extensions/amp-bind/amp-bind.md

```diff
@@ -66,425 +72,972 @@ The `amp-bind` component allows you to add custom stateful interactivity to your
     width="480" height="270"></amp-youtube>
 <figcaption>Watch this video for an introduction to amp-bind.</figcaption></figure>
 
-### A simple example
+`amp-bind` has three main concepts:
 
-In the following example, tapping the button changes the `<p>` element's text from "Hello World" to "Hello amp-bind".
+1. [State](#state): A document-scope, mutable JSON state. State variables update in response to user actions. `amp-bind` does not evaluate expressions on page load. Visual elements should have their default "state" defined and not rely `amp-bind` for initial render.
+2. [Expressions](#expressions): JavaScript-like expressions that can reference the **state**.
+3. [Bindings](#bindings): Special attributes that link an element's property to a **state** via an **expression**. A property is bound by wrapping it inside brackets, in the form of `[property]`.
+
+### Basic example
+
+[example preview="inline" playground="true" imports="amp-bind"]
 
 ```html
 <p [text]="'Hello ' + foo">Hello World</p>
 
-<button on="tap:AMP.setState({foo: 'amp-bind'})">Say "Hello amp-bind"</button>
+<button on="tap:AMP.setState({foo: 'Interactivity'})">
+  Say "Hello Interactivity"
+</button>
 ```
 
-{% call callout('Note', type='note') %}
-For performance and to avoid the risk of unexpected content jumping, `amp-bind` does not evaluate expressions on page load. This means that the visual elements should be given a default state and not rely `amp-bind` for initial render.
-{% endcall %}
+[/example]
 
-### How does it work?
+In the example above:
 
-`amp-bind` has three main components:
+- The **state** begins as empty.
+- It has a single **binding** to `[text]`, the text content of a node, on the `<p>` element.
+- The `[text]` value contains the **expression**, `'Hello ' + foo`. This expression concatenates the string 'Hello ' and the value of the **state variable** foo.
 
-1. [State](#state): A document-scope, mutable JSON state. In the example above, the state is empty before tapping the button. After tapping the button, the state is `{foo: 'amp-bind'}`.
-2. [Expressions](#expressions): These are JavaScript-like expressions that can reference the **state**. The example above has a single expression, `'Hello ' + foo`, which concatenates the string literal `'Hello '` and the state variable `foo`.
-   There is a limit of 100 operands what can be used in an expression.
-3. [Bindings](#bindings): These are special attributes of the form `[property]` that link an element's property to an **expression**. The example above has a single binding, `[text]`, which updates the `<p>` element's text every time the expression's value changes.
+When the user taps/clicks the button:
 
-`amp-bind` takes special care to ensure speed, security and performance on AMP pages.
+1. It triggers the `tap` event.
+1. The `tap` event invokes the `AMP.setState()` method.
```

Comment:
Are these ok to be all the number 1?

Reply:
Markdown converts them into their proper numbers :)